### PR TITLE
API review for CoreWebView2WebAuthenticationRequested event

### PR DESCRIPTION
**Pull request by yildirimcagri**
Added WebAuthenticationRequested.md API spec file.

### CODE REVIEW

**Comment by david-risney** [comment]
The initial review PR should go from user branch A -> user branch B. Then a new PR with API review board from user branch B -> master.

File: specs/WebAuthenticationRequested.md [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+      [in] ICoreWebView2StagingWebAuthenticationRequestedEventArgs* args);
+}
+
+/// Represents a Basic HTTP authentication response that contains a user name
+/// and a password as according to RFC7617 (https://tools.ietf.org/html/rfc7617)
+[uuid(bc9cfd60-29c4-4943-a83b-d0d2f3e7df03), object, pointer_default(unique)]
+interface ICoreWebView2StagingWebAuthenticationResponse : IUnknown
+{
+  /// User name provided for authorization.
+  [propget] HRESULT UserName([out, retval] LPWSTR* userName);
```
Comment (david-risney) [inline specs/WebAuthenticationRequested.md:197]: Are there any restrictions on what's allowed in the username or password fields? If so, what happens if the end dev provides invalid values?

File: specs/WebAuthenticationRequested.md [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+  3) Ask the user for credentials via the default login prompt 
+We also propose CoreWebView2StagingWebAuthenticationResponse, the runtime class that represents the app's response with credentials to the basic authentication request.
+
+# Examples
+## Basic usage
+CoreWebView2WebAuthenticationRequestedEvent follows the same event handling pattern as any other WebView2 event.
+
+To add event handler:
+```cpp
+webviewStaging2->add_WebAuthenticationRequested(
```
Comment (david-risney) [inline specs/WebAuthenticationRequested.md:40]: We want to use the interface names we expect for the public API rather than staging

**Review by david-risney — CHANGES_REQUESTED** [review]

File: specs/WebAuthenticationRequested.md (hunk 1 above)
Comment (yildirimcagri) [reply specs/WebAuthenticationRequested.md:197]: I had referenced the RFC link for the specification of the username password above. Do we need to specify any behavior for the case the developer doesn't go by specification?

**Pull request closed without merging**